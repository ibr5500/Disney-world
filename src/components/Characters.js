import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchData, charactersFilter } from '../redux/characters/characters';

function Characters() {
  const displayCharacters = useSelector((state) => state.characters);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!displayCharacters.length) dispatch(fetchData());
  });

  const [searchItem, setSearchItem] = useState('');

  const handleSearch = (e) => {
    if (e.searchItem === '' || e.searchItem === ' ') {
      dispatch(fetchData().sort());
    }
    dispatch(charactersFilter(searchItem));
    setSearchItem(e.target.value);
  };

  return (
    <>
      <div className="input-container">
        <input
          type="text"
          placeholder="Search ..."
          onChange={handleSearch}
          value={searchItem}
        />
      </div>
      <div className="container">
        {displayCharacters.sort().map((character, index) => (
          <div className={(index % 2 === 0) ? 'row-bg card' : 'card'} key={character.key}>
            <Link to={{ pathname: `/${character.id}` }} className="details-link">
              <img className="character-img" src={character.image} alt={character.name} />
              {' '}
              <h3 className="character-name">
                {(character.name[0] === '.' || character.name[0] === "'")
                  ? character.name.replace(/[.']/g, '')
                  : character.name}
              </h3>
            </Link>
          </div>
        ))}
      </div>
    </>
  );
}

export default Characters;
