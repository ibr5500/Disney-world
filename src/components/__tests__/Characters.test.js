import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Characters from '../Characters';

describe('Test for Characters page', () => {
  it('Render the Characters', () => {
    const tree = renderer.create(<Provider store={store}><Characters /></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
