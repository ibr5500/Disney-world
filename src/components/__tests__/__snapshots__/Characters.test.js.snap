// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Characters page Render the Characters 1`] = `
<div
  className="container"
/>
`;

exports[`Test for Characters page Should render the Characters 1`] = `
<div
  className="container"
/>
`;
